ui/src/app/login/page.tsx

```diff
@@ -0,0 +1,64 @@
+"use client"
+
+import { login } from "../queries"
+import { useMutation } from "react-query"
+import { useState, ChangeEvent } from "react"
+import { useCookies } from "react-cookie"
+import { useRouter } from "next/navigation"
+
+export default function LoginPage() {
```

Comment:
Can we move the login box to the middle of the page (instead of being at the top)?

Reply:
done (this was the most challenging thing I've done)